gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,478 @@
+        assert res is None, (
+            f"Function returned {res} for product_category {product_category},"
+            f" expected None"
+        )
+
+    # Test a product_category that doesn't exist
+    for product_category in ['NOT', 'A', 'CATEGORY']:
+        with pytest.raises(ValueError):
+            checks.check_product_category(product_category)
+
+
+def test_check_product_level(
+    product_levels: list[str],
+) -> None:
+    ''' Test check_product_level() '''
+
+    # Test types that aren't strings
+    for product_level in [123, None]:
+        with pytest.raises(ValueError):
+            checks.check_product_level(product_level)
+
+    # Test a product_level that does exist
+    for product_level in product_levels:
+        res = checks.check_product_level(product_level)
+        assert res is None, (
+            f"Function returned {res} for product_level {product_level}, "
+            f"expected None"
+        )
+
+    # Test a product_level that doesn't exist
+    for product_level in ['NOT', 'A', 'LEVEL']:
+        with pytest.raises(ValueError):
+            checks.check_product_level(product_level)
+
+
+def test_check_product_validity(
+    product_types: list[str],
+) -> None:
+    """ Test check_product_validity() """
+
+    # Test a product that does exist
+    for product_type in product_types:
+        for product in available_products(product_type=product_type):
+            res = checks.check_product_validity(product,
+                                                product_type=product_type)
+            assert res is None, (
+                f"Function returned {res} for product {product}, "
+                f"expected None"
+            )
+
+    # Test a product that doesn't exist
+    for product_type in product_types:
+        for product in [('IMERG'), 123, None]:
+            with pytest.raises(ValueError):
+                checks.check_product_validity(product,
+                                              product_type=product_type)
+            # Test a None product type
+            with pytest.raises(ValueError):
+                checks.check_product_validity(product, product_type=None)
+
+
+def test_check_time() -> None:
+    ''' Test that time is returned a datetime object from varying inputs '''
+
+    # Test a string
+    res = checks.check_time('2014-12-31')
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31)
+
+    # Test a datetime object
+    res = checks.check_time(datetime.datetime(2014, 12, 31))
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31)
+
+    # Test a datetime timestamp with h/m/s/ms
+    res = checks.check_time(datetime.datetime(2014, 12, 31, 12, 30, 30, 300))
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31, 12, 30, 30, 300)
+
+    # Test a np.datetime64 object of "datetime64[s]"
```

Comment:
Can we test it works if np.datetime 64 is `ns`. Should be able to cast as `ms`

Reply:
Yes I expected that. Modify `check_time` that if `np.array` or `np.datetime64`, first change resolution to `ms`, and then convert to datetime

Reply:
@evanjt